### ISSUE TITLE
Disabled due to no economy bug

### ISSUE THREAD

**Issue by InflamedSebi** [leothawne/LTItemMail#1]
I use Vault (1.7.1) and Skyowallet (1.2.1) in Spigot (1.13), but LTItemMail says:
```
[04:12:15] [Server thread/INFO]: ( ( *
--
[04:12:15] [Server thread/INFO]: )\ ) * ) )\ ) ) ( ` (
[04:12:15] [Server thread/INFO]: (()/` ) /( (()/(( /( ( ) )\))( ) ( )\
[04:12:15] [Server thread/INFO]: /(_)( )(_)) /(_))\())))\ ( ((_)()\ ( /( )\((_)
[04:12:15] [Server thread/INFO]: (_))(_(_()) (_))(_))//((_) )\ ' (_()((_))(_)((_)_ V 0.2.1 (Minecraft 1.13.X)
[04:12:15] [Server thread/INFO]: \| \| \|_ _\| \|_ _\| \|_(_)) _((_)) \| \/ ((_)_ (_\| \| Works with Java 8+
[04:12:15] [Server thread/INFO]: \| \|__ \| \| \| \|\| _/ -_\| ' \() \| \|\/\| / _` \|\| \| \| Released on 16/11/2018 (America/Sao_Paulo)
[04:12:15] [Server thread/INFO]: \|____\|\|_\| \|___\|\__\___\|_\|_\|_\| \|_\| \|_\__,_\|\|_\|_\| Twitter @leothawne
[04:12:15] [Server thread/INFO]:
[04:12:15] [Server thread/INFO]: [LT Item Mail INFO] Loading mailboxes...
[04:12:15] [Server thread/INFO]: [LT Item Mail INFO] Loading Vault...
[04:12:15] [Server thread/INFO]: [LT Item Mail ERROR] An economy plugin is missing! Disabling LT Item Mail...
[04:12:15] [Server thread/INFO]: [LTItemMail] Disabling LTItemMail v0.2.1
[04:12:15] [Server thread/INFO]: [LT Item Mail INFO] Closing mailboxes...
[04:12:15] [Server thread/INFO]: ( ( )
[04:12:15] [Server thread/INFO]: )\ ) )\ ) ( /( ( (
[04:12:15] [Server thread/INFO]: (()/( ( ( (()/( )\()))\ ) ))\
[04:12:15] [Server thread/INFO]: /(_))_ )\ )\ ((_)((_)\(()/( /((_)
[04:12:15] [Server thread/INFO]: (_)) __\|((_)((_) _\| \|\| \|(_))(_)(_))
[04:12:15] [Server thread/INFO]: \| (_ / _ / _ / _` \|\| '_ \| \|\| / -_)
[04:12:15] [Server thread/INFO]: \___\___\___\__,_\|\|_.__/\_, \___\|
[04:12:15] [Server thread/INFO]: \|__/
```
all other plugins that use economy via vault are fine with skyowallet.

**Comment by Nobookie** [comment]
Thank you for your report! I'll start to fix it right now.

**Comment by Nobookie** [comment]
**The possible cause of this error:** The plugin is been disabled because the server cannot register the Economy class (Economy.class). And that appears to be a Vault error. **I can assure you that this error is not because of my code.** Here it is:
`public boolean isReady() {
	rsp = plugin.getServer().getServicesManager().getRegistration(Economy.class);
	if(rsp != null) {
		return true;
	}
	return false;
}`
**According to this (https://github.com/MilkBowl/VaultAPI/#implementing-vault)** there is nothing I can do to fix it. I cannot change the Vault API code to fix this problem.
**All I can do** is recommend you to re-download and replace the Vault jar file and/or look for another Economy plugin and see if it works fine.

### **To the next versions: Vault will no longer be a required dependency. I'll make it optional.**

**Comment by Nobookie** [comment]
A tip from me: [EssentialsX](https://ci.ender.zone/job/EssentialsX/) is a good Economy plugin.
Documentation: https://github.com/EssentialsX/Essentials/wiki

**Comment by InflamedSebi** [comment]
Switched to GemsEconomy, since essentials is too overkill for my server.
I use a Quickshop remake too, which uses:
```RegisteredServiceProvider<Economy>` economyProvider = Bukkit.getServicesManager().getRegistration(Economy.class);
		if (economyProvider != null) {...```

I don't think `Bukkit.getServicesManager()` differs from `plugin.getServer().getPluginManager()` but thats the only difference i can see ...

Or it might be a race condition? can you make sure your plugin is loaded last? So other economy plugins have a chance to register in vault before you check them?

**Comment by InflamedSebi** [comment]
Just checked. Log clearly shows your plugin was enabled first so the Economy plugin did not register yet.
If you're testing with EssentialsX it might be just lucky your plugin is loading after it?

**Comment by Nobookie** [comment]
**I just thought about that too!** For some reason my plugin is been enabled before the Economy plugin. I really don't know how to fix this. I would need to list every Economy plugin that uses Vault, and there's so many of them... Do you have any suggestion?

**Comment by Nobookie** [comment]
**Perhaps my plugin could be loaded after all worlds available are loaded by the server itself.** What do you think?
This could give some time to the Economy plugins be loaded first.

**Comment by InflamedSebi** [comment]
Well it's been a while since i did mc plugins ... but back at 1.7 there was a plugin.yml where you could list dependencies and the startup time ... usually economy providers had a [soft]depend on vault and a load: startup. I checked gemseconomy and it has neigther ... so it is probably his fault ...

you could not make your plugin wait, except when using dependencies .... there was a rather bad workaround where you put "zzzzz" as softdepend and this made your plugin load last since it waited for zzzzz ... but I wouldn't do it now ... It might not even work anymore ... imma just add the startup to his plugin yml and test again ... I'll report back tomorrow.

**Comment by Nobookie** [comment]
See the current **plugin.yml**:
![plugin](https://user-images.githubusercontent.com/31146142/49046527-97a2b000-f1bb-11e8-8f5a-d27e305e34a7.png)
Vault is listed in "depend" and not in "soft-depend".
Also I changed "load" value from STARTUP to POSTWORLD.

**Comment by Nobookie** [comment]
@InflamedSebi **I did some tests with Skyowallet** and **I released a new version (0.2.2)**. I think you might want use it on your server! xD
https://www.spigotmc.org/resources/lt-item-mail.62294/update?update=254131

**Comment by leothawne** [comment]
# Bug fixed!
